app/models/concerns/master_file_behavior.rb

```diff
@@ -68,7 +68,7 @@ def stream_details(token,host=nil)
   end
 
   def embed_title
-    "#{ self.media_object.title } - #{ self.title || self.file_location.split( "/" ).last }"
+    "#{ self.media_object.title } - #{ self.title.present? ? self.title : self.file_location.split( "/" ).last }"
```

Comment:
This is another place where a nil file_location could bite us, but since it is only if title is also blank then I think it is  even more of a corner case.